pkg/ext-proc/backend/pod_controller.go

```diff
@@ -0,0 +1,49 @@
+package backend
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+)
+
+type PodController struct {
```

Comment:
why do we still need this? The EndpointSlice reconciler will be the one looking up the endpoints and storing them in the datastore